src/Domain/Creators/TimeDependence/UniformRotationAboutZAxis.hpp

```diff
@@ -0,0 +1,159 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/Rotation.hpp"
+#include "Domain/Creators/TimeDependence/GenerateCoordinateMap.hpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain {
+namespace FunctionsOfTime {
+class FunctionOfTime;
+}  // namespace FunctionsOfTime
+namespace CoordinateMaps {
+namespace TimeDependent {
+template <typename Map1, typename Map2>
+class ProductOf2Maps;
+}  // namespace TimeDependent
+}  // namespace CoordinateMaps
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+class CoordinateMap;
+}  // namespace domain
+
+namespace Frame {
+struct Grid;
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+/*!
+ * \brief A uniform rotation about the \f$z\f$ axis:
+ * \f{eqnarray*}
+ * x &\to& x \cos \alpha(t) - y \sin \alpha(t) \\
+ * y &=\to& y \sin \alpha(t) + y \cos \alpha(t)
+ * \f},
+ * where \f$\alpha(t)\f$ is a `FunctionOfTime`. For 3 spatial dimensions,
+ * \f$z \to z\f$, and the translation is implemented as a product of the 2D
+ * rotation and an identity map. This time dependence is the identity for
+ * 1 spatial dimension.
```

Comment:
I did it this way because I didn't know how to get it to work with 1D being undefined. I'll remove this sentence and implement your suggested way to deal with this, because I agree that really a 1D rotation should not be defined or implemented.